### ISSUE TITLE
yarn rebuild (Feature request)

### ISSUE THREAD

**Issue by rstacruz** [yarnpkg/yarn#2069]
Npm has `npm rebuild` to rebuild compiled packages (eg, [node-sass](https://www.npmjs.com/package/node-sass)). It would be nice if yarn has this, too.

**Comment by alanhogan** [comment]
Our use-case for this: Copying a project into a docker container which doesn't have permissions to pull all of our packages. A rebuild is required. (And `npm rebuild` seems to error on the `node_modules` left by `yarn` for some reason.)

**Comment by rstacruz** [comment]
possibly related: https://github.com/yarnpkg/yarn/issues/756

**Comment by joshrickert** [comment]
Occasionally I have issues after switching environments around that are solved by an `npm rebuild`. It's one of the few tasks I still have to rely on npm for currently.

**Comment by pgnd** [comment]
fyi, with

```
yarn --version
	0.24.5
node -v
	v8.1.2
npm -v
	5.0.3
```
after 

```
yarn add --dev node-sass ...
```
exec of

```
npm rebuild node-sass
```

fails with

```
> node-sass@4.5.3 install /dev/www/yarn-test/node_modules/node-sass
> node scripts/install.js

module.js:487
    throw err;
    ^

Error: Cannot find module '/dev/www/yarn-test/node_modules/node-sass/scripts/install.js'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-sass@4.5.3 install: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-sass@4.5.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-06-15T17_14_17_946Z-debug.log
```

but checking, the file does exist

```
ls /dev/www/yarn-test/node_modules/node-sass/scripts/install.js
	/dev/www/yarn-test/node_modules/node-sass/scripts/install.js
```

**Comment by olalonde** [comment]
Dupe of https://github.com/yarnpkg/yarn/issues/756

**Comment by BYK** [comment]
Closing in favor of #756.

**Comment by ctcpip** [comment]
@BYK please reopen this issue.  The other issue resolves in context of node version changing.  We need an actual rebuild package command for cross-platform considerations.

**Comment by aahoughton** [comment]
Agreed, as commented in #4750.

**Comment by giggio** [comment]
Yes, those are different requests. This should be reopened.

**Comment by loziniak** [comment]
Is there a workaround?

**Comment by aahoughton** [comment]
Depends on what your core problem is -- there is no equivalent for `npm rebuild` due to architecture changes, which is a use case a number of us have.  The `npm rebuild` due to node version changes fix is what -- possibly mistakenly -- lead to this issue being closed.

@BYK can we get confirmation that this issue won't be re-opened, and we should file a new one?

**Comment by loziniak** [comment]
> > Is there a workaround?

> Depends on what your core problem is

Looking at all the comments at the topic, I think something like most of people - I changed node version and needed to rebuild _node-sass_. I managed to do that by using `yarn install --force`.

**Comment by shaunc** [comment]
I have a docker image that I build in stages, first excluding dev dependencies; later installing and building tests (finally, if tests run, taking original node_modules for production.). For test install, I need to use `npm rebuild sass`. Other than that, I could leave out npm from my image entirely...

**Comment by fugroup** [comment]
I used `npm rebuild --update-binary` to resolve my issues. It doesn't create a package-lock file either so it shouldn't interfere with yarn.

**Comment by masaeedu** [comment]
@loziniak doesn't seem like `yarn install --force` does anything. I had to rebuild the platform-folders module after upgrading my Node JS version, but couldn't find any way to do this using Yarn (only after I used `npm rebuild` as @BobbyWibowo shows above did it work). Not sure why this issue and the "closed in favor of" issue are both closed.

**Comment by masaeedu** [comment]
For visibility, I found: https://github.com/yarnpkg/yarn/issues/5271, which is what I guess folks trying to use Yarn with xplat modules need to keep an eye on.
